File: exercises/day13-Keyboard-First-Refactoring/scala/src/main/scala/sleightrip/exception/Exceptions.scala
package sleightrip.exception

/**
 * Thrown when no elf is logged into the workshop system.
 */
class ElfNotLoggedInException 
  extends RuntimeException("No elf is logged into the workshop system")

/**
 * Thrown when a static collaborator (like UserSession or TripDAO) is called directly.
 */
class CollaboratorCallException(message: String) extends RuntimeException(message)

File: exercises/day13-Keyboard-First-Refactoring/typescript/src/exception/Exceptions.ts
/**
 * Thrown when no elf is logged into the workshop system.
 */
export class ElfNotLoggedInException extends Error {
    constructor() {
        super('No elf is logged into the workshop system');
        this.name = 'ElfNotLoggedInException';
    }
}

/**
 * Thrown when a static collaborator (like UserSession or TripDAO) is called directly.
 */
export class CollaboratorCallException extends Error {
    constructor(message: string) {
        super(message);
        this.name = 'CollaboratorCallException';
    }
}
